### DISCUSSION TITLE
Why useFluentProviderThemeStyleTag does not use useId from @fluentui/react-hooks

### DISCUSSION THREAD

**Discussion by flvyu** [microsoft/fluentui#26447] [Q&A] [answered]
I noticed that the hook useFluentProviderThemeStyleTag is implemented with the `useId` hook from `@fluentui/react-utilities`. However, this `useId` does not store the id in the global scope which I have seen causes id collisions causing bugs such as Themes being overwritten.

I notice other parts of the codebase uses `useId` from @fluentui/react-hooks which **does** store the id in the global scope. 

Is it okay to update useFluentProviderThemeStyleTag to be implemented using useId from @fluentui/react-hooks?

**Comment by layershifter** [comment, marked as answer]
@flvyu 

- `@fluentui/react-utilities` is a package for v9 components
- `@fluentui/react-hooks` is a package for v8 components

Why global scope is not a good solution I commented in the issue itself https://github.com/microsoft/fluentui/issues/26448#issuecomment-1400031788.

**Reply by flvyu** [reply]
Thanks for following up!
